### ISSUE TITLE
Cypress: Unexpected token on nullish coalescence operator

### ISSUE THREAD

**Issue by andresz1** [mswjs/headers-polyfill#58]
Hello, I'm getting this error with `@cypress/webpack-preprocessor` due to this [change](https://github.com/mswjs/headers-polyfill/commit/cff8faf4f92ba8cfc0e62d000ecde4eb78a0a491#diff-8c0d76939cb7407404fdc980ee64cc4a48d814d77abc5f0f2edcd83153c3d723R74)
- cypress version `10.4.0`
```
Error: Webpack Compilation Error
../node_modules/msw/node_modules/headers-polyfill/lib/index.js 105:64
Module parse failed: Unexpected token (105:64)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
|   }
|   get(name) {
>     return this[NORMALIZED_HEADERS][normalizeHeaderName(name)] ?? null;
|   }
|   set(name, value) {
 @ ../node_modules/msw/lib/node/index.mjs 269:0-51 276:22-37 411:0-43 414:22-29 488:0-71 491:0-51 494:18-33 570:26-42 642:0-55 696:19-27 720:0-55 723:0-55 755:19-27
 @ ../node_modules/@s-ui/mock/lib/server.js
 @ ../node_modules/@s-ui/mock/lib/index.js
 @ ../mocks/index.js
 @ ./support/index.js
    at handle (/home/travis/.cache/Cypress/10.4.0/Cypress/resources/app/node_modules/@packages/server/node_modules/@cypress/webpack-preprocessor/dist/index.js:180:23)
    at finalCallback (/home/travis/.cache/Cypress/10.4.0/Cypress/resources/app/node_modules/webpack/lib/Compiler.js:257:39)
    at /home/travis/.cache/Cypress/10.4.0/Cypress/resources/app/node_modules/webpack/lib/Compiler.js:306:14
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/home/travis/.cache/Cypress/10.4.0/Cypress/resources/app/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:6:1)
    at AsyncSeriesHook.lazyCompileHook (/home/travis/.cache/Cypress/10.4.0/Cypress/resources/app/node_modules/tapable/lib/Hook.js:154:20)
    at /home/travis/.cache/Cypress/10.4.0/Cypress/resources/app/node_modules/webpack/lib/Compiler.js:304:22
    at Compiler.emitRecords (/home/travis/.cache/Cypress/10.4.0/Cypress/resources/app/node_modules/webpack/lib/Compiler.js:499:39)
    at /home/travis/.cache/Cypress/10.4.0/Cypress/resources/app/node_modules/webpack/lib/Compiler.js:298:10
    at /home/travis/.cache/Cypress/10.4.0/Cypress/resources/app/node_modules/webpack/lib/Compiler.js:485:14
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/home/travis/.cache/Cypress/10.4.0/Cypress/resources/app/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:6:1)
    at AsyncSeriesHook.lazyCompileHook (/home/travis/.cache/Cypress/10.4.0/Cypress/resources/app/node_modules/tapable/lib/Hook.js:154:20)
    at /home/travis/.cache/Cypress/10.4.0/Cypress/resources/app/node_modules/webpack/lib/Compiler.js:482:27
    at /home/travis/.cache/Cypress/10.4.0/Cypress/resources/app/node_modules/neo-async/async.js:2818:7
    at done (/home/travis/.cache/Cypress/10.4.0/Cypress/resources/app/node_modules/neo-async/async.js:3522:9)
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/home/travis/.cache/Cypress/10.4.0/Cypress/resources/app/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:6:1)
    at AsyncSeriesHook.lazyCompileHook (/home/travis/.cache/Cypress/10.4.0/Cypress/resources/app/node_modules/tapable/lib/Hook.js:154:20)
    at /home/travis/.cache/Cypress/10.4.0/Cypress/resources/app/node_modules/webpack/lib/Compiler.js:464:33
    at /home/travis/.cache/Cypress/10.4.0/Cypress/resources/app/node_modules/@packages/server/node_modules/graceful-fs/graceful-fs.js:143:16
    at /home/travis/.cache/Cypress/10.4.0/Cypress/resources/app/node_modules/@packages/server/node_modules/graceful-fs/graceful-fs.js:143:16
    at /home/travis/.cache/Cypress/10.4.0/Cypress/resources/app/node_modules/@packages/server/node_modules/graceful-fs/graceful-fs.js:61:14
    at FSReqCallback.oncomplete (node:fs:192:23)
```

**Comment by evanwills** [comment]
Here is a repo to demonstrate the issue: https://github.com/evanwills/msw-header-pollyfill--webpack-error/

**Comment by kettanaito** [comment]
Thank you, @evanwills 🙌 The best. 

I think it comes down to us shipping unprocessed modern syntax (the nullish coalescence operator `??`). We likely need to set the proper `target` in our `tsconfig.json` to force those expressions to be transpiled into common JavaScript.

**Comment by evanwills** [comment]
While this is being fixed, here's a temporary workaround for others who have this issue:

Add the following to the your package.json
```json
  "overrides": {
    "headers-polyfill": "3.1.2"
  }
```
This will force use of the older version of headers-polyfill that doesn't have this issue. Keep an eye on this issue so you can remove the `overrides...` block when it's resolved.

**Comment by jordevo** [comment]
thank you @evanwills , adding the `overrides` to our project's `package.json` worked for us.

**Comment by kettanaito** [comment]
I've checked our TypeScript config and we are emitting for `es5` which should be fine. I looked into the emitted code and noticed that `??` doesn't get transpiled. I then looked at the [support map for nullish coalescing](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing#browser_compatibility), which is supported in all major browsers for three years now. It's also supported in Node.js since version 14, which is already EOL. 

Can you please try upgrading your version of Cypress and see if it solved the issue? It looks like a Cypress/webpack-preprocessor issue, the JavaScript `headers-polyfill` is shipping should run everywhere. I've noticed you are using Cypress 10 which is more than a year old. Perhaps this has been addressed in Cypress 13, which is the latest stable version? 

Please let me know. Thanks.

**Comment by giroberto** [comment]
Hi, just adding information for the troubleshooting, but I fixed overriding the version as suggested in a comment above. 
I have a project created and still using an old version of create-react-app(^3.4.4), and I am having the same problem starting the app. (headers-polyfill is a dependency of msw in my case)

**Comment by felipeplets** [comment]
I was able to test locally and create a PR that fixes the issue: https://github.com/mswjs/headers-polyfill/pull/66

**Comment by kettanaito** [comment]
Fixed by #66.

**Comment by tmartin2089** [comment]
@lcorniglione did you ever figure out the cause?
